### PR TITLE
add markdown style

### PR DESCRIPTION
**Pull request by weakiwi**
add markdown table style support

### CODE REVIEW

File: prettytable.py [hunk 1]
```diff
@@ -73,6 +73,7 @@
 MSWORD_FRIENDLY = 11
 PLAIN_COLUMNS = 12
 RANDOM = 20
+MARKDOWN = 28
```
Comment (hugovk) [inline prettytable.py]: Any particular reason for putting this at the end and choosing `28`, and not putting it after `PLAIN_COLUMNS`?

File: prettytable_test.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 import unittest
 from math import e, pi, sqrt
 
-from prettytable import (ALL, MSWORD_FRIENDLY, NONE, PrettyTable, from_csv,
+from prettytable import (ALL, MSWORD_FRIENDLY, MARKDOWN, NONE, PrettyTable, from_csv,
```
Comment (hugovk) [inline prettytable_test.py]: It doesn't look like Markdown is actually being tested here?

File: prettytable.py (hunk 1 above)
Comment (weakiwi) [reply prettytable.py]: You are correct. Let me fix this

File: prettytable_test.py (hunk 2 above)
Comment (weakiwi) [reply prettytable_test.py]: Let me add test case

File: README.md [hunk 3]
```diff
@@ -369,6 +369,7 @@ you can use for your tables:
 made
   * `PLAIN_COLUMNS` - A borderless style that works well with command line
 programs for columnar data
+  * `MARKDOWN` - A style that follows markdown syntax
```
Comment (hugovk) [inline README.md]:
```suggestion
  * `MARKDOWN` - A style that follows Markdown syntax
```

File: prettytable.py [hunk 4]
```diff
@@ -975,9 +976,21 @@ def set_style(self, style):
             self._set_columns_style()
         elif style == RANDOM:
             self._set_random_style()
+        elif style == MARKDOWN:
+            self._set_markdown_style()
```
Comment (hugovk) [inline prettytable.py:978]: Minor thing, let's also move this block before `RANDOM`

File: prettytable_test.py [hunk 5]
```diff
@@ -636,6 +638,19 @@ def testHtmlOutputFormated(self):
...
 
+class MarkdownStyleTest(BasicTests):
+    def testMarkdownStyle(self):
+        t = PrettyTable(['Field 1', 'Field 2', 'Field 3'])
+        t.add_row(['value 1', 'value2', 'value3'])
+        t.add_row(['value 4', 'value5', 'value6'])
+        t.add_row(['value 7', 'value8', 'value9'])
+        t.set_style(MARKDOWN)
+        result = t.get_string()
+        assert result.strip() == """| Field 1 | Field 2 | Field 3 |
```
Comment (hugovk) [inline prettytable_test.py]: Add a newline for readability and consistency with the others
```suggestion
        assert result.strip() == """
| Field 1 | Field 2 | Field 3 |
```

File: prettytable_test.py [hunk 6]
```diff
@@ -636,6 +638,19 @@ def testHtmlOutputFormated(self):
...
+|---------|---------|---------|
+| value 1 |  value2 |  value3 |
+| value 4 |  value5 |  value6 |
+| value 7 |  value8 |  value9 |""".strip()
```
Comment (hugovk) [inline prettytable_test.py]:
```suggestion
| value 7 |  value8 |  value9 |
""".strip()
```

**Review by hugovk — CHANGES_REQUESTED** [review]
A couple of little suggestions.

Also, to fix the [lint check on the CI](https://github.com/jazzband/prettytable/pull/19/checks?check_run_id=510020506), please run:

```sh
pip install -U black isort
black .
isort -y
```
Thanks!

File: prettytable_test.py [hunk 7]
```diff
@@ -637,6 +638,25 @@ def testHtmlOutputFormated(self):
...
+    def testMarkdownStyle(self):
+        t = PrettyTable(["Field 1", "Field 2", "Field 3"])
+        t.add_row(["value 1", "value2", "value3"])
+        t.add_row(["value 4", "value5", "value6"])
+        t.add_row(["value 7", "value8", "value9"])
+        t.set_style(MARKDOWN)
+        result = t.get_string()
+        assert (
+            result.strip()
+            == """| Field 1 | Field 2 | Field 3 |
```
Comment (hugovk) [inline prettytable_test.py]:
```suggestion
            == """
| Field 1 | Field 2 | Field 3 |
```

**Review by weakiwi — COMMENTED** [review]
all fix

**Review by hugovk — APPROVED** [review]

**Comment by hugovk** [comment]
Thanks!

**Pull request merged**